From 29ad7f1a6935371b1145354be92f758cdbc37e74 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:01 -0400
Subject: [ethernet] i40e/i40evf: Change type to u32 to avoid sparse error

Message-id: <1409658042-22689-121-git-send-email-sassmann@redhat.com>
Patchwork-id: 90316
O-Subject: [RHEL7.1 PATCH 120/281] i40e/i40evf: Change type to u32 to avoid sparse error
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Greg Rose <gregory.v.rose@intel.com>

tx_lpi_status and rx_lpi_status are declared as bool but then used in sizeof
operations in the CORE driver code.  Make them u32 to make sparse be quiet.

Change-ID: Iad6daeb1c7149e61ece242acd18c64b320c246a3
Signed-off-by: Greg Rose <gregory.v.rose@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 10bc478a2acca55875f32bdaeaa60d2ad60ced62)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_type.h b/drivers/net/ethernet/intel/i40e/i40e_type.h
index 0fe158c..71f9718 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_type.h
+++ b/drivers/net/ethernet/intel/i40e/i40e_type.h
@@ -1016,8 +1016,8 @@ struct i40e_hw_port_stats {
  u64 mac_short_packet_dropped; /* mspdc */
  u64 checksum_error;  /* xec */
  /* EEE LPI */
- bool tx_lpi_status;
- bool rx_lpi_status;
+ u32 tx_lpi_status;
+ u32 rx_lpi_status;
  u64 tx_lpi_count;  /* etlpic */
  u64 rx_lpi_count;  /* erlpic */
 };
-- 
1.7.1